app/commands/exercise/representation/create_or_update.rb

```diff
@@ -3,23 +3,64 @@ class Representation
     class CreateOrUpdate
       include Mandate
 
-      initialize_with :submission, :ast, :ast_digest, :mapping, :last_submitted_at
+      initialize_with :submission, :ast, :ast_digest, :mapping, :representer_version, :exercise_version, :last_submitted_at
 
       def call
-        representation = Exercise::Representation.find_create_or_find_by!(exercise:, ast_digest:) do |rep|
+        # First cache the old representation
+        @old_representation = Exercise::Representation.where(source_submission: submission).order('id DESC').first
```

Comment:
I think you could also do:

```suggestion
        @old_representation = Exercise::Representation.where(source_submission: submission).last
```

This (without the submission clause) generates the following SQL:

```sql
SELECT `exercise_representations`.* FROM `exercise_representations` ORDER BY `exercise_representations`.`id` DESC LIMIT 1
```